From b0d1b46dd45e71b2409daa309771259b9cccbbdd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A5rten=20Kongstad?= <marten.kongstad@sonymobile.com>
Date: Thu, 2 Jun 2016 09:34:36 +0200
Subject: [PATCH 10/33] OMS-N: Fix memory leak during idmap creation [10/14]

Plug a memory leak in AssetManager::createIdmap.

Bug: 31052947

AOSP Change-Id: I83af3a40516ed2d50d5a7c8ee175ed960fde9933

Conflicts:
	libs/androidfw/AssetManager.cpp

Change-Id: I46d2127a80bbfb1f928341f514d67017b68e5f32
---
 libs/androidfw/AssetManager.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libs/androidfw/AssetManager.cpp b/libs/androidfw/AssetManager.cpp
index 4097089..9b559d1 100644
--- a/libs/androidfw/AssetManager.cpp
+++ b/libs/androidfw/AssetManager.cpp
@@ -299,7 +299,10 @@ bool AssetManager::createIdmap(const char* targetApkPath, const char* overlayApk
                 ALOGW("failed to find resources.arsc in %s\n", ap.path.string());
                 goto exit;
             }
-            tables[i].add(assets[i]);
+            if (tables[i].add(assets[i]) != NO_ERROR) {
+                ALOGW("failed to add %s to resource table", paths[i].string());
+                goto exit;
+            }
         }
         ret = tables[0].createIdmap(tables[1], targetCrc, overlayCrc,
                 targetApkPath, overlayApkPath, (void**)outData, outSize) == NO_ERROR;
-- 
2.10.0

